core/controllers/editor.py

```diff
@@ -683,10 +683,12 @@ def get(self, exploration_id):
             raise self.PageNotFoundException
 
         # Patch `snapshots` to use the editor's display name.
+        snapshots_committer_ids = [
+            snapshot['committer_id'] for snapshot in snapshots]
+        committer_usernames = user_services.get_usernames(
+            snapshots_committer_ids)
         for snapshot in snapshots:
-            if snapshot['committer_id'] != feconf.SYSTEM_COMMITTER_ID:
-                snapshot['committer_id'] = user_services.get_username(
-                    snapshot['committer_id'])
+            snapshot['committer_id'] = committer_usernames.pop(0)
```

Comment:
I think popping the first element is O(N)? Maybe just do "for index, snapshot in enumerate(snapshots)" and set snapshot['committer_id'] to committer_usernames[index].

How are you planning to handle the SYSTEM_COMMITTER_ID case?

Reply:
Oh yup right, thanks :)
And the `SYSTEM_COMMITTER_ID` case is handled in `get_users_settings` function that's called inside `get_usernames` in user_services.py
Though one thing that is confusing me is, in `get_username` function `MIGRATION_BOT_USER_ID` case is handled separately and `get_usernames` doesn't consider it ?

Reply:
Ah, that seems incorrect -- the two functions should have the same behaviour. Would you mind making updates so that they each handle all the cases?

Reply:
Ofcourse, I'll update! Just to confirm - `SYSTEM_COMMITTER_ID` case should be handled inside `get_usernames` function?

Reply:
Yep!

Reply:
Done.